### PR TITLE
Sdk nightly build

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25606
## Description
Added some conditions to the nightly build and created a steps summary file which is used in the slack notifier script.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -138,21 +159,37 @@ jobs:
 
             [ -n "${BACKWARD_COMPATIBILITY}" ] && CHECK_BACKWARD="false" || CHECK_BACKWARD="true"
             ./Tests/scripts/validate.sh
+
+            if [ $? -ne 0 ] && [ -n "${DEMISTO_SDK_NIGHTLY}" ] ;
```
Comment (yaakovi) [inline .circleci/config.yml:163]: need to make sure that if the flake8 fails, we also fail the step
in this state - I'm not sure this is true.

File: Tests/scripts/slack_notifier.py [hunk 2]
```diff
@@ -34,44 +34,49 @@ def options_handler():
...
     return options
 
 
-def get_failing_unit_tests_file_data():
+def get_failing_entities_file_data(report_file_name):
     try:
-        failing_ut_list = None
-        file_name = './artifacts/failed_lint_report.txt'
+        failing_entities_list = None
+        file_name = "./artifacts/{}.txt".format(report_file_name)
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:48]: f-string please (in all places)

File: Tests/scripts/slack_notifier.py [hunk 3]
```diff
@@ -34,44 +34,49 @@ def options_handler():
...
+            with open(file_name, 'r') as failed_entities_file:
+                failing_entity = failed_entities_file.readlines()
+                failing_entities_list = [line.strip('\n') for line in failing_entity]
         else:
-            print('Did not find failed_lint_report.txt file')
+            print("Did not find {}.txt file".format(report_file_name))
     except Exception as err:
-        print_error('Error getting failed_lint_report.txt file: \n {}'.format(err))
-    return failing_ut_list
+        print_error("Error getting {}.txt file: \n {}".format(report_file_name, err))
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:57]: when can we expect to get an exception? I think we can remove this

File: Tests/scripts/slack_notifier.py [hunk 4]
```diff
@@ -202,10 +223,40 @@ def slack_notifier(build_url, slack_token, env_results_file_name, container):
         )
 
 
+def sdk_slack_notifier(build_url, slack_token, container):
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:226]: very similar to `slack_notifier` function.
add an argument for `is_sdk_nightly` there and change `env_results_file_name` to optional

Comment (reutshal) [reply Tests/scripts/slack_notifier.py]: Done

File: .circleci/config.yml (hunk 1 above)
Comment (reutshal) [reply .circleci/config.yml]: Done

File: Tests/scripts/slack_notifier.py [hunk 5]
```diff
@@ -204,41 +204,20 @@ def slack_notifier(build_url, slack_token, env_results_file_name, container):
         print("Extracting build status")
         # container 1: unit tests
         if int(container):
-            print_color("Starting Slack notifications about nightly build - unit tests", LOG_COLORS.GREEN)
-            content_team_attachments = get_attachments_for_unit_test(build_url)
+            print_color(f'Starting Slack notifications about {build_type} build - unit tests', LOG_COLORS.GREEN)
+            if is_sdk_nightly:
+                content_team_attachments = get_attachments_for_unit_test(build_url, is_sdk_build=True)
+            else:
+                content_team_attachments = get_attachments_for_unit_test(build_url)
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:211]:
```suggestion
            content_team_attachments = get_attachments_for_unit_test(build_url, is_sdk_build=is_sdk_nightly)
```

File: Tests/scripts/slack_notifier.py [hunk 6]
```diff
@@ -254,9 +233,9 @@ def sdk_slack_notifier(build_url, slack_token, container):
 def main():
     options = options_handler()
     if options.nightly:
-        slack_notifier(options.url, options.slack, options.env_results_file_name, options.node_index)
+        slack_notifier(options.url, options.slack, options.node_index, 'nightly', options.env_results_file_name)
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:236]:
```suggestion
        slack_notifier(options.url, options.slack, options.node_index, 'nightly', 
            env_results_file_name =options.env_results_file_name)
```

File: Tests/scripts/slack_notifier.py [hunk 7]
```diff
@@ -12,6 +13,28 @@
...
 
 
+def get_faild_steps_list():
+    options = options_handler()
+    failed_steps_list = []
+    circle_client = circle_api(options.circleci)
+    vcs_type = 'github'
+    build_report = circle_client.get_build_info(username='demisto', project='content', build_num=options.buildNumber,
+                                                vcs_type=vcs_type)
+    for step in build_report.get('steps', ''):
```
Comment (yaakovi) [inline Tests/scripts/slack_notifier.py:23]:
```suggestion
    for step in build_report.get('steps', []):
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -122,6 +126,23 @@ jobs:
...
+                  python3 -m pylint --errors-only ./Tests
+                  pylint_exit_code=$?
+
+                  if [ $pylint_exit_code -ne 0 ] ; then
+                    echo "Infrastructure testing - pylint" >> $CIRCLE_ARTIFACTS/task_report.txt
+                  fi
+
+                let "exit_code = $flake8_exit_code + $validate_exit_code"
+                exit $exit_code
+            fi
```
Comment (yaakovi) [inline .circleci/config.yml:145]: looks like left-overs from the original approach. please update

File: .circleci/config.yml [hunk 9]
```diff
@@ -122,6 +126,13 @@ jobs:
             python3 -m pytest ./Tests/Marketplace/Tests/marketplace_services_test.py -v
             python3 -m pytest ./Tests/Marketplace/Tests/upload_packs_test.py -v
             python3 -m pytest ./Tests/Marketplace/Tests/search_and_install_packs_test.py -v
+
+            if [ -n "${DEMISTO_SDK_NIGHTLY}" ] ; then
+                  echo Checking for sdk related pylint errors
```
Comment (yaakovi) [inline .circleci/config.yml:131]: move inside the bash script

File: .circleci/config.yml [hunk 10]
```diff
@@ -122,6 +126,13 @@ jobs:
...
+                  ./Tests/scripts/sdk_pylint_check.sh
+            fi
+            exit 0
+            fi
```
Comment (yaakovi) [inline .circleci/config.yml:135]: remove

File: .circleci/config.yml [hunk 11]
```diff
@@ -664,3 +675,15 @@ workflows:
     jobs:
       - build:
           context: nightly_env
+  sdk_nightly:
+    triggers:
+      - schedule:
+          # should trigger every day at 9 PM UTC (12 AM Israel Time)
```
Comment (yaakovi) [inline .circleci/config.yml:726]: this is 12 AM UTC which is 3 AM Israel time (also fix the comment in the regular nightly)

**Pull request merged**